### PR TITLE
refactor(switch): refactor some switch with match

### PR DESCRIPTION
**Pull request by helyakin**
| Q             | A
| ------------- | ---
| Branch?       |  main

I just wanted to refactor some switch with match to ease the readability.

### CODE REVIEW

**Comment by helyakin** [comment]
Can I have some review for this refacto @soyuka or @alanpoulain ?
That would be very nice :)

**Comment by alanpoulain** [comment]
LGTM

File: src/Metadata/Extractor/ResourceExtractorTrait.php [hunk 1]
```diff
@@ -53,22 +53,19 @@ private function phpize(\SimpleXMLElement|array|null $resource, string $key, str
...
-            case 'string':
-                return (string) $resource[$key];
-            case 'integer':
-                return (int) $resource[$key];
-            case 'bool':
+        return match ($type) {
+            'bool|string' => \is_bool($resource[$key]) || \in_array((string) $resource[$key], ['1', '0', 'true', 'false'], true) ? $this->phpize($resource, $key, 'bool') : $this->phpize($resource, $key, 'string'),
+            'string' => (string) $resource[$key],
+            'int' => (int) $resource[$key],
+            'bool' => function () use ($resource, $key) {
```
Comment (alanpoulain) [inline src/Metadata/Extractor/ResourceExtractorTrait.php:60]: Shouldn't the function be called here?

Comment (helyakin) [reply src/Metadata/Extractor/ResourceExtractorTrait.php]: Well done, you are right !

**Comment by soyuka** [comment]
can you rebase ? I fixed the test suite so it should be mostly green

**Comment by helyakin** [comment]
@soyuka rebase has been done 🫡

Thanks for your time !

**Review by alanpoulain — APPROVED** [review]

**Comment by soyuka** [comment]
This has failing tests

**Pull request closed without merging**